test/auto_parallel/pir/test_static_pir_program.py

```diff
@@ -141,6 +144,66 @@ def test_build_with_shard_tensor(self):
         self.assertEqual(dist_w1.dist_attr().process_mesh, mesh)
         self.assertEqual(dist_w1.dist_attr().dims_mapping, [-1, 0])
 
+    def test_build_with_apply_mix2dist_pass(self):
+        paddle.enable_static()
+        with paddle.pir_utils.IrGuard():
+            main_program = paddle.base.Program()
+            with paddle.base.program_guard(main_program):
+                mesh = dist.ProcessMesh([0, 1], dim_names=['dp'])
+                input1 = paddle.randint(low=0, high=1000, shape=[8, 4])
+                output1 = dist.shard_tensor(input1, mesh, [dist.Shard(0)])
+
+                input2 = paddle.randn([4, 8])
+                output2 = dist.shard_tensor(input2, mesh, [dist.Shard(1)])
+
+                self.assertTrue(input1.is_dense_tensor_type())
+                self.assertTrue(input2.is_dense_tensor_type())
+
+        self.assertTrue(main_program.num_ops() == 6)
+
+        self.assertFalse(input1.use_empty())
+        self.assertFalse(input2.use_empty())
+
+        self.assertTrue(output1.use_empty())
+        self.assertTrue(output2.use_empty())
+
+        self.assertFalse(input1.get_defining_op().has_attr("op_dist_attr"))
+        self.assertFalse(input2.get_defining_op().has_attr("op_dist_attr"))
+
+        # check dist type
+        self.assertTrue(output1.is_dist_dense_tensor_type())
+        self.assertTrue(output2.is_dist_dense_tensor_type())
+
+        # run apply_mix2dist_pass
+        apply_mix2dist_pass(main_program)
+
+        # after apply_mix2dist_pass, the program changed
+        self.assertTrue(main_program.num_ops() == 4)
+
+        self.assertTrue(input1.is_dist_dense_tensor_type())
+        self.assertTrue(input2.is_dist_dense_tensor_type())
+
+        self.assertTrue(input1.get_defining_op().has_attr("op_dist_attr"))
+        self.assertTrue(input2.get_defining_op().has_attr("op_dist_attr"))
+
```

Comment:
the operations appended implicitly by rand-related-op should be verified 

in your case here, should check the dist_attr for :
 input1.get_defining_op.operand[1] 
and 
input1.get_defining_op.operand[1].defining_op()

Reply:
thanks, already added